products-service/src/main/java/com/petclinic/products/datalayer/products/ProductRepository.java

```diff
@@ -22,6 +24,7 @@ public interface ProductRepository extends ReactiveMongoRepository<Product, Stri
     @Query("{ 'productSalePrice' : { $lte: ?0 } }")
     Flux<Product> findByProductSalePriceLessThanEqual(Double maxPrice);
     Flux<Product> findProductsByProductType(String productType);
+    List<Product> findByProductType(ProductType productType);
```

Comment:
Convert the List to a Flux<Product>, that way you can use Reactive streams instead of manually creating them in the controller